kube-core/src/metadata.rs

```diff
@@ -23,39 +23,57 @@ pub struct TypeMeta {
 /// schema without knowing the details of the version.
 #[derive(Deserialize, Serialize, Clone, Default, Debug)]
 #[serde(rename_all = "camelCase")]
-pub struct PartialObjectMeta {
+pub struct PartialObjectMeta<K> {
     /// The type fields, not always present
     #[serde(flatten, default)]
     pub types: Option<TypeMeta>,
     /// Standard object's metadata
     #[serde(default)]
     pub metadata: ObjectMeta,
+    /// Type information for static dispatch
+    #[serde(skip, default)]
+    pub _phantom: PhantomData<K>,
 }
 
-impl From<PartialObjectMeta> for ObjectMeta {
-    fn from(obj: PartialObjectMeta) -> Self {
+// Users usually want the inner metadata on returns
+impl<K> From<PartialObjectMeta<K>> for ObjectMeta {
+    fn from(obj: PartialObjectMeta<K>) -> Self {
         ObjectMeta { ..obj.metadata }
     }
 }
 
-impl Resource for PartialObjectMeta {
-    type DynamicType = ApiResource;
-    type Scope = DynamicResourceScope;
+// Unit tests often convert the other way
+impl<K> From<ObjectMeta> for PartialObjectMeta<K> {
```

Comment:
Is this a valid thing in general, or do we want to make it a dedicated associated function where we can call out any caveats? Should we take the `TypeMeta` from `K` instead of making up our own?

Reply:
Ugh, this was a really good call. Thank you.  The `From` is way too simplistic here because how we want to use `PartialObjectMeta` depends on whether it is a request object or a response object (the apiserver erases the type, but the root type is needed for patch).

Have removed the `From` and instead created a sealed `PartialObjectMetaExt` trait that contains two converter functions, and have used it in the main test case.